src/service.js

```diff
@@ -450,19 +464,48 @@ function getServicePromiseInternal(holder, id) {
   // TODO(@cramforce): Add a check that if the element is eventually registered
   // that the service is actually provided and this promise resolves.
   const deferred = new Deferred();
-  const {promise, resolve} = deferred;
+  const {promise, resolve, reject} = deferred;
 
   const services = getServices(holder);
   services[id] = {
     obj: null,
     promise,
     resolve,
+    reject,
     context: null,
     ctor: null,
   };
   return promise;
 }
 
+/**
+ * @param {!Object} holder
+ * @param {string} id of the service.
+ * @param {string|number} error
```

Comment:
Can we make this an `!Error`? Otherwise will have to type-check `.catch(value => ...)`.

Reply:
changed to `{*}` to be more generic.